File: app/src/main/java/io/github/howiezuo/unsplash/api/ApiComponent.java
package io.github.howiezuo.unsplash.api;

import javax.inject.Singleton;

import dagger.Component;
import io.github.howiezuo.unsplash.feature.detail.DetailViewModel;
import io.github.howiezuo.unsplash.feature.main.MainViewModel;

@Singleton
@Component(
        modules = {
                ApiModule.class
        }
)
public interface ApiComponent {

    void inject(MainViewModel viewModel);

    void inject(DetailViewModel viewModel);

}

File: app/src/main/java/io/github/howiezuo/unsplash/api/ApiComponent.kt
package io.github.howiezuo.unsplash.api

import dagger.Component
import io.github.howiezuo.unsplash.feature.detail.DetailViewModel
import io.github.howiezuo.unsplash.feature.main.MainViewModel
import javax.inject.Singleton


@Singleton
@Component(
        modules = arrayOf(
                ApiModule::class
        )
)
interface ApiComponent {

    fun inject(vm: MainViewModel)

    fun inject(vm: DetailViewModel)

}
